File: gaspi-standard/examples/hello_world.c
#include <stdio.h>
#include <stdlib.h>
#include <GASPI.h>

int
main (int argc, char *argv[])
{
  gaspi_proc_init (GASPI_BLOCK);

  gaspi_rank_t iProc;
  gaspi_rank_t nProc;

  gaspi_proc_rank (&iProc);
  gaspi_proc_num (&nProc);

  printf ("Hello world from rank %i of %i!\n", iProc, nProc);

  gaspi_proc_term (GASPI_BLOCK);

  return EXIT_SUCCESS;
}

File: gaspi-standard/examples/hello_world.f90
program hello_world

  use gaspi_c_binding
  implicit none
  integer(gaspi_return_t) :: res
  integer(gaspi_rank_t) :: rank, num
  integer(gaspi_timeout_t) :: timeout

  timeout = GASPI_BLOCK

  res = gaspi_proc_init(timeout)
  res = gaspi_proc_rank(rank)
  res = gaspi_proc_num(num)

  print *,"Hello world from rank ",rank

  res = gaspi_proc_term(timeout)

end program hello_world
